### PR TITLE
feat: put strict upperbound version for rattler-build-conda-compat

### PR DESCRIPTION
**Pull request by nichmor**
Checklist
* [x] Added a ``news`` entry
* [ ] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

### CODE REVIEW

**Comment by beckermr** [comment]
We should submit a repodata patch with this fix as well.

File: environment.yml [hunk 1]
```diff
@@ -36,4 +36,4 @@ dependencies:
   - jsonschema
   - backports.strenum
   - exceptiongroup
-  - rattler-build-conda-compat >=0.2.4
+  - rattler-build-conda-compat >=0.2.4, <1
```
Comment (beckermr) [inline environment.yml:39]:
```suggestion
  - rattler-build-conda-compat >=0.2.4,<1.0.0a0
```

**Review by beckermr — CHANGES_REQUESTED** [review]
Please add a news item for this one.

**Comment by beckermr** [comment]
xref: https://github.com/conda-forge/conda-smithy-feedstock/issues/300

**Comment by nichmor** [comment]
@beckermr added news entry here: https://github.com/conda-forge/conda-smithy/pull/2009/files#diff-8f0dc6783ef019e72b814a18ca111bdc937349f343e9217d38220d9e9cdd90b3R7

File: news/2008-rbcc-upper-bound.rst [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+* <news item>
+
+**Fixed:**
+
+* <news item>
+
+**Security:**
+
+* <news item>
```
Comment (beckermr) [inline news/2008-rbcc-upper-bound.rst:23]:
```suggestion
**Added:**

* <news item>

**Changed:**

* <news item>

**Deprecated:**

* <news item>

**Removed:**

* <news item>

**Fixed:**

* Add upper bound for rattler-build-conda-compat. (#2009)

**Security:**

* <news item>
```

**Comment by nichmor** [comment]
hey @beckermr - are there anything else left to do for this PR? thanks!

**Pull request merged**